lib/mongo/retryable.rb

```diff
@@ -163,6 +208,52 @@ def write_with_retry(session, write_concern, ending_transaction = false, &block)
 
     private
 
+    def modern_read_with_retry(session, server_selector, &block)
+      attempt = 0
+      server = select_server(cluster, server_selector)
+      begin
+        yield server
+      rescue Error::SocketError, Error::SocketTimeoutError => e
+        if session.in_transaction?
+          raise
+        end
+        retry_read(e, server_selector, &block)
+      rescue Error::OperationFailure => e
+        if session.in_transaction? || !e.write_retryable?
```

Comment:
Should this be `!e.read_retryable?`

Reply:
No, per the spec in https://github.com/mongodb/specifications/blob/master/source/retryable-reads/retryable-reads.rst#retryable-error the set of retryable errors is the same for reads and writes, hence there is no `read_retryable?` method to invoke.

Reply:
The [documentation for `write_retryable?`](https://github.com/mongodb/mongo-ruby-driver/blob/ba8b70edbd660c4ec6cefd6718140aa33427b9a0/lib/mongo/error/operation_failure.rb#L93) says "Can the write operation that caused the error be retried?", and the [documentation for `retryable?`](https://github.com/mongodb/mongo-ruby-driver/blob/ba8b70edbd660c4ec6cefd6718140aa33427b9a0/lib/mongo/error/operation_failure.rb#L81) says "Can the read operation that caused the error be retried?". It's pretty confusing that we wouldn't use the method specifically documented to be used for read retries in the method that retries reads. Can you elaborate on why using `retryable?` wouldn't work?

Reply:
`retryable?` deals with the legacy retryable read criteria and as a result is not applicable to modern retryable reads or writes.

Reply:
This PR should update the documentation on that method then, since it's not at all clear from the current language

Reply:
Made the requested changes.

Reply:
Thanks!